app/controllers/saved_pages_controller.rb

```diff
@@ -0,0 +1,46 @@
+class SavedPagesController < ApplicationController
+  include AuthenticatedApiConcern
+
+  before_action :check_saved_page_exists, only: %i[destroy show]
+
+  # GET /api/saved_pages
+  def index
+    render_api_response(saved_pages: user.saved_pages.map(&:to_hash))
+  end
+
+  # GET /api/saved_pages/:page_path
+  def show
+    render_api_response(saved_page: saved_page.to_hash)
+  end
+
+  # POST /api/saved_pages
+  def create
+    saved_page = SavedPage.find_or_create_by(oidc_user_id: user.id, page_path: params[:page_path])
+
+    if saved_page.persisted?
+      render_api_response(saved_page: saved_page.to_hash)
+    else
+      raise ApiError::CannotSavePage, { page_path: params[:page_path], errors: saved_page.errors }
+    end
+  end
+
+  # DELETE /api/saved_pages
+  def destroy
+    saved_page.destroy!
+    head :no_content
+  end
+
+private
+
+  def user
+    @user ||= @govuk_account_session.user
+  end
+
+  def saved_page
+    @saved_page ||= SavedPage.find_by(oidc_user_id: user.id, page_path: params[:page_path])
+  end
+
+  def check_saved_page_exists
+    raise ApiError::SavedPageNotFound, { page_path: params[:page_path] } if saved_page.nil?
+  end
```

Comment:
A thought: is this new error type actually valuable?  What does it let a client app do that a regular 404 does not?

- `UnknownAttributeNames` gives a list of the unknown attributes, which is useful, as the client can request several attributes at once and only some may be invalid.  That's not the case here, as the client only requests a single saved page.
- `LevelOfAuthenticationTooLow` returns the level that the client needs to reauthenticate the user at, otherwise the client would have to guess.  That's not the case here, as the error doesn't explain how to fix the problem.
- `CannotSavePage` returns the list of validation errors, which could be logged or shown to the user, letting them fix the problem themselves (though, probably logged rather than shown).  Here there's just one error: this page hasn't been saved.

So, I'm leaning towards this being just a plain body-less 404 response.

Reply:
This is an optional suggestion